src/main/java/androidx/widget/TextView.kt

```diff
@@ -0,0 +1,156 @@
+ * to just set one or more Drawables.
+ *
+ * @see TextView.setCompoundDrawablesWithIntrinsicBounds
+ */
+fun TextView.updateCompoundDrawablesWithIntrinsicBounds(
+    left: Drawable? = drawableSentinel,
+    top: Drawable? = drawableSentinel,
+    right: Drawable? = drawableSentinel,
+    bottom: Drawable? = drawableSentinel
+) {
+    val drawables = updateDrawables(left, top, right, bottom, compoundDrawables)
+    setCompoundDrawablesWithIntrinsicBounds(drawables[0], drawables[1], drawables[2], drawables[3])
+}
+
+/**
+ * Updates this TextView's Drawables. This version of the method allows using named parameters
+ * to just set one or more Drawables.
+ *
+ * @see TextView.setCompoundDrawablesRelative
+ */
+@RequiresApi(17)
+fun TextView.updateCompoundDrawablesRelative(
+    start: Drawable? = drawableSentinel,
+    top: Drawable? = drawableSentinel,
+    end: Drawable? = drawableSentinel,
+    bottom: Drawable? = drawableSentinel
+) {
+    val drawables = updateDrawables(start, top, end, bottom, compoundDrawablesRelative)
+    setCompoundDrawablesRelative(drawables[0], drawables[1], drawables[2], drawables[3])
+}
+
+/**
+ * Updates this TextView's Drawables. This version of the method allows using named parameters
+ * to just set one or more Drawables. Use 0 if you do not want a Drawable there.
+ *
+ * @see TextView.setCompoundDrawablesRelativeWithIntrinsicBounds
+ */
+@RequiresApi(17)
+fun TextView.updateCompoundDrawablesRelativeWithIntrinsicBounds(
+    @DrawableRes start: Int = -1,
+    @DrawableRes top: Int = -1,
+    @DrawableRes end: Int = -1,
+    @DrawableRes bottom: Int = -1
+) {
+    val compoundDrawablesRelative = compoundDrawablesRelative
+    setCompoundDrawablesRelativeWithIntrinsicBounds(
+        getUpdatedCompoundDrawable(start, compoundDrawablesRelative[0]),
+        getUpdatedCompoundDrawable(top, compoundDrawablesRelative[1]),
+        getUpdatedCompoundDrawable(end, compoundDrawablesRelative[2]),
+        getUpdatedCompoundDrawable(bottom, compoundDrawablesRelative[3])
+    )
+}
+
+/**
+ * Updates this TextView's Drawables. This version of the method allows using named parameters
+ * to just set one or more Drawables.
+ *
+ * @see TextView.setCompoundDrawablesRelativeWithIntrinsicBounds
+ */
+@RequiresApi(17)
+fun TextView.updateCompoundDrawablesRelativeWithIntrinsicBounds(
+    start: Drawable? = drawableSentinel,
+    top: Drawable? = drawableSentinel,
+    end: Drawable? = drawableSentinel,
+    bottom: Drawable? = drawableSentinel
+) {
+    val drawables = updateDrawables(start, top, end, bottom, compoundDrawablesRelative)
+    setCompoundDrawablesRelativeWithIntrinsicBounds(
+        drawables[0], drawables[1], drawables[2], drawables[3]
+    )
+}
+
+private fun updateDrawables(
+    start: Drawable?, top: Drawable?, end: Drawable?, bottom: Drawable?, srcArray: Array<Drawable>
+): Array<Drawable?> {
+    val drawables = lazy(LazyThreadSafetyMode.NONE) { srcArray }
```

Comment:
Note that this is useless here as `srcArray` (and `compoundDrawables`) will already have been called (and the array created). Here again, inlining avoid the issue.

Reply:
Note that you could pass the `Lazy` instead.

Reply:
Point 1) I need it for the Lazy instance.

Point 2) Indeed, but that would mean that in four call sites we'd have `lazy(NONE) { compoundDrawablesRelative }` or `lazy(NONE) { compoundDrawables}` instead of `compoundDrawablesRelative` and `compoundDrawables` . 

If these changes are required I'm happy to apply them.